java/client/src/main/java/glide/managers/CommandManager.java

```diff
@@ -31,36 +39,46 @@ public <T> CompletableFuture<T> submitNewCommand(
         // when complete, convert the response to our expected type T using the given responseHandler
         return channel
                 .write(prepareRedisRequest(command.getRequestType(), command.getArguments()), true)
-                .thenApplyAsync(response -> responseHandler.apply(response));
+                .thenApplyAsync(responseHandler::apply);
+    }
+
+    /**
+     * Build a command and send.
+     *
+     * @param command The command to execute
+     * @param route The routing options
+     * @param responseHandler The handler for the response object
+     * @return A result promise of type T
+     */
+    public <T> CompletableFuture<T> submitNewCommand(
```

Comment:
Is it better to using Optional(Route) and avoid overloading and duplicating code?